### PR TITLE
[CAPR] Add new provisioning tests for etcd snapshot creation/restore, encryption key rotation, and certificate rotation.

### PR DESCRIPTION
**Pull request by Oats87**
## Issue:
https://github.com/rancher/rancher/issues/41125
https://github.com/rancher/rancher/issues/41080

### CODE REVIEW

**Review by HarrisonWAffel — APPROVED** [review]

File: tests/integration/pkg/cluster/clusters.go [hunk 1]
```diff
@@ -286,6 +316,37 @@ func getPodLogs(clients *clients.Clients, podNamespace, podName string) (string,
...
+		podNamespace,
+		"cp",
+		fmt.Sprintf("%s:%s", podName, podPath),
+		destFile,
+	}
+
+	cmd := exec.Command("kubectl", kcp...)
+	if err := cmd.Run(); err != nil {
+		logrus.Errorf("error running kubectl -n %s cp %s:%s %s", podNamespace, podName, podPath, destFile)
+		return "", nil
```
Comment (bfbachmann) [inline tests/integration/pkg/cluster/clusters.go:333]: Should an error be returned here?
```suggestion
		logrus.Errorf("error running kubectl -n %s cp %s:%s %s", podNamespace, podName, podPath, destFile)
		return "", err
```

If so, do we need the log message, or can we leave it to the caller to log if necessary?

File: tests/integration/pkg/tests/custom/custom_operations_test.go [hunk 2]
```diff
@@ -0,0 +1,475 @@
...
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/util/retry"
+)
+
+func TestCustomEtcdSnapshotOperations(t *testing.T) {
+	clients, err := clients.New()
+	if err != nil {
+		t.Fatal(err)
+	}
```
Comment (bfbachmann) [inline tests/integration/pkg/tests/custom/custom_operations_test.go:33]: Nit: could also do this for brevity
```suggestion
	assert.NoError(t, err)
```

File: tests/integration/pkg/tests/custom/custom_operations_test.go [hunk 3]
```diff
@@ -0,0 +1,475 @@
...
+		if apierrors.IsNotFound(err) || err == nil {
+			return true
+		}
+		return false
+	},
+		func() error {
+			snapshots, err := clients.RKE.ETCDSnapshot().List(c.Namespace, metav1.ListOptions{})
+			if err != nil || len(snapshots.Items) == 0 {
+				return err
+			}
```
Comment (bfbachmann) [inline tests/integration/pkg/tests/custom/custom_operations_test.go:142]: Not sure if it's important but should this return an error if `len(snapshots.Items) == 0`? I figure we want to retry if that slice is empty, but I don't think will if `err` is `nil`. Might have to do something like

```go
if err != nil {
    return err
} else if len(snapshots.Items) == 0 {
    return errors.New("no items found in snapshot")
}
```

**Review by bfbachmann — COMMENTED** [review]
Looks good for the most part. Just minor comments.

**Review by jakefhyde — APPROVED** [review]

File: tests/integration/pkg/tests/custom/custom_operations_test.go [hunk 4]
```diff
@@ -0,0 +1,475 @@
...
+	"github.com/rancher/rancher/tests/integration/pkg/systemdnode"
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/util/retry"
+)
+
+func TestCustomEtcdSnapshotOperations(t *testing.T) {
```
Comment (bfbachmann) [inline tests/integration/pkg/tests/custom/custom_operations_test.go:29]: Maybe this is a nit as well (sorry haha 😬 ), but I notice that these tests share a lot of setup code. For instance, all of them initialize clients, spin up and RKE cluster (control plane, worker, etcd), wait for it to be ready, initialize a wrangler context, and so on. Would it be worth moving any of that code into some shared helper or setup function?

**Review by bfbachmann — COMMENTED** [review]
Feel free to ignore my comments if you need this to me merged soon. They're mostly cosmetic.

**Pull request merged**